app/tx.go

```diff
@@ -0,0 +1,44 @@
+
+package app
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/x/cash"
+	"github.com/iov-one/weave/x/multisig"
+	"github.com/iov-one/weave/x/sigs"
+)
+
+// TxDecoder creates a Tx and unmarshals bytes into it
+func TxDecoder(bz []byte) (weave.Tx, error) {
+	tx := new(Tx)
+	err := tx.Unmarshal(bz)
+	if err != nil {
+		return nil, err
+	}
+	return tx, nil
+}
+
+// make sure tx fulfills all interfaces
+var _ weave.Tx = (*Tx)(nil)
```

Comment:
Which tx's dex tx must fulfill?

Reply:
I do not understand the question.

If you are asking about this construct, I think the Go documentation can explain this much better than me https://golang.org/doc/faq#implements_interface

Reply:
Okay, I'll expand the question. As you can see in [bnsd/tx.go](https://github.com/iov-one/weave/blob/master/cmd/bnsd/app/codec.proto#L29-L65) all the tx messages supported by bnsd are imported in codec. Also, I can see that in [mycoind example](https://github.com/iov-one/weave/blob/v0.16.0/examples/mycoind/app/codec.proto#L9-L22). Which tx messages should dex module have?

Reply:
Unless you are having special needs, you should include all messages exposed by the extension you are using. If you don't include a message from an extension, you most likely lose some functionality that processing of this message was enabling.
Does this make sense?